<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1070</width>
    <height>618</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>1070</width>
    <height>618</height>
   </size>
  </property>
  <property name="sizeIncrement">
   <size>
    <width>1070</width>
    <height>618</height>
   </size>
  </property>
  <property name="palette">
   <palette>
    <active>
     <colorrole role="WindowText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Button">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>226</green>
        <blue>228</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Light">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Midlight">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>240</green>
        <blue>241</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Dark">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>127</red>
        <green>113</green>
        <blue>114</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Mid">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>170</red>
        <green>151</green>
        <blue>152</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Text">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="BrightText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ButtonText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Base">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Window">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>226</green>
        <blue>228</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Shadow">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="AlternateBase">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>240</green>
        <blue>241</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ToolTipBase">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>220</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ToolTipText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="PlaceholderText">
      <brush brushstyle="SolidPattern">
       <color alpha="128">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
    </active>
    <inactive>
     <colorrole role="WindowText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Button">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>226</green>
        <blue>228</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Light">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Midlight">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>240</green>
        <blue>241</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Dark">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>127</red>
        <green>113</green>
        <blue>114</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Mid">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>170</red>
        <green>151</green>
        <blue>152</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Text">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="BrightText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ButtonText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Base">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Window">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>226</green>
        <blue>228</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Shadow">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="AlternateBase">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>240</green>
        <blue>241</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ToolTipBase">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>220</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ToolTipText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="PlaceholderText">
      <brush brushstyle="SolidPattern">
       <color alpha="128">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
    </inactive>
    <disabled>
     <colorrole role="WindowText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>127</red>
        <green>113</green>
        <blue>114</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Button">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>226</green>
        <blue>228</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Light">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Midlight">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>240</green>
        <blue>241</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Dark">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>127</red>
        <green>113</green>
        <blue>114</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Mid">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>170</red>
        <green>151</green>
        <blue>152</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Text">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>127</red>
        <green>113</green>
        <blue>114</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="BrightText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>255</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ButtonText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>127</red>
        <green>113</green>
        <blue>114</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Base">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>226</green>
        <blue>228</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Window">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>226</green>
        <blue>228</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="Shadow">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="AlternateBase">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>226</green>
        <blue>228</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ToolTipBase">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>255</red>
        <green>255</green>
        <blue>220</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="ToolTipText">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
     <colorrole role="PlaceholderText">
      <brush brushstyle="SolidPattern">
       <color alpha="128">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </colorrole>
    </disabled>
   </palette>
  </property>
  <property name="windowTitle">
   <string>Алгоритм плавающего горизонта</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>-10</y>
      <width>431</width>
      <height>181</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QComboBox" name="funcs">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>401</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
    </widget>
    <widget class="QLineEdit" name="x_end_lineEdit">
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>80</y>
       <width>60</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>5</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="z_end_lineEdit">
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>130</y>
       <width>60</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>5</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>320</x>
       <y>80</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>dx:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>320</x>
       <y>130</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>dz:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="delta_z_lineEdit">
     <property name="geometry">
      <rect>
       <x>360</x>
       <y>130</y>
       <width>59</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>0.2</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="delta_x_lineEdit">
     <property name="geometry">
      <rect>
       <x>360</x>
       <y>80</y>
       <width>59</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>0.2</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="z_start_lineEdit">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>130</y>
       <width>59</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>-5</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="x_start_lineEdit">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>80</y>
       <width>59</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>-5</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>80</y>
       <width>71</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Xmax</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>130</y>
       <width>71</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Zmax</string>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>80</y>
       <width>71</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Xmin</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>130</y>
       <width>71</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Zmin</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>530</y>
      <width>431</width>
      <height>81</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QPushButton" name="draw_surface_pushButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>201</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Построить</string>
     </property>
    </widget>
    <widget class="QPushButton" name="clear_pushButton">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>30</y>
       <width>201</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Очистить</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>410</y>
      <width>431</width>
      <height>131</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>20</y>
       <width>271</width>
       <height>61</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Коэффициент
масштабирования</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>411</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Масштабировать</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="k_lineEdit">
     <property name="geometry">
      <rect>
       <x>300</x>
       <y>30</y>
       <width>61</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>10</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>160</y>
      <width>431</width>
      <height>261</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QPushButton" name="right_pushButton">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>130</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Вправо</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="autoRepeat">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="left_pushButton">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>130</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Влево</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="autoRepeat">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="up_pushButton">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>80</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Вверх</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="autoRepeat">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="down_pushButton">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>170</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Вниз</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="autoRepeat">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="oz_pushButton">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>120</y>
       <width>121</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Oсь OZ</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="autoRepeat">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLineEdit" name="rotate_ox_lineEdit">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>30</y>
       <width>64</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>1</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>12</x>
       <y>30</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>OX:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="rotate_oy_lineEdit">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>30</y>
       <width>65</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>1</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>30</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>OY:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>30</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>OZ:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="rotate_oz_lineEdit">
     <property name="geometry">
      <rect>
       <x>330</x>
       <y>30</y>
       <width>61</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>1</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>230</y>
       <width>351</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Пробел - повернуть</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="draw_widget" native="true">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>10</y>
      <width>600</width>
      <height>600</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color:lightgray;</string>
    </property>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
